src/ResourceManager/ContainerRegistry/Commands.ContainerRegistry/Commands/TestAzureContainerRegistryWebhook.cs

```diff
@@ -0,0 +1,72 @@
+﻿// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using System.Management.Automation;
+using Microsoft.Azure.Management.ContainerRegistry.Models;
+
+namespace Microsoft.Azure.Commands.ContainerRegistry
+{
+    [Cmdlet(VerbsDiagnostic.Test, ContainerRegistryWebhookNoun, DefaultParameterSetName = ResourceIdParameterSet)]
+    [OutputType(typeof(EventInfo))]
```

Comment:
@Jinhuafei avoid exposing your SDK types in the public API. I would wrap this `EventInfo` class in a `PSEventInfo` class, or return a `bool`

Reply:
why can't we expose SDK types?

Reply:
@Jinhuafei exposing SDK types makes it more difficult to mitigate breaking changes that may arise in the future.

For example, if you were to, for some reason, remove the `Id` property of the `EventInfo` type, the following script would no longer work:

```powershell
$name = Get-RandomName
$resourceGroup = "testRG"
$registryName = "testRegistry"
$result = Test-AzureRmContainerRegistryWebhook -Name $name -ResourceGroupName $resourceGroup -RegistryName $registryName
if ($result -ne $null)
{
    Write-Host "SUCCESS: Id --> $($result.Id)"
}
```

If you create a type that wraps an SDK type, then you can remove the property in the SDK, publish the SDK to NuGet, and update the PS type to bring back the property and let users know that it will be removed in an upcoming breaking change release.

Reply:
I see. Will fix this. From the other side, you can always add a wrap class whenever the SDK type changes.